### PR TITLE
Story4 details page

### PR DESCRIPTION
**Pull request by MrLukey**
Created DetailedPCGameEntity, added getGameById Hydrator method, added $id to PCGameAbstract, created gameDetails.php file and styled inline with main styling

### CODE REVIEW

File: gameDetails.php [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+require 'vendor/autoload.php';
+use GameListing\Hydrators\GamesHydrator;
+
+$db = new PDO('mysql:host=db;dbname=games', 'root', 'password');
+$gameId = $_GET['id'];
```
Comment (norayna) [inline gameDetails.php:9]: need to validate this in some way or handle what happens if the id doesn't exist - currently it does this:
image

File: src/Hydrators/GamesHydrator.php [hunk 2]
```diff
@@ -2,13 +2,23 @@
...
+        $query = $db->query('SELECT `title`, `genre`, `thumbnail`, `id` FROM `pc-games`;');
         $query->setFetchMode(\PDO::FETCH_CLASS, PCGameEntity::class);
         return $query->fetchAll();
     }
+
+    public static function getGameById(\PDO $db, int $id): DetailedPCGameEntity
+    {
+        $query = $db->query('SELECT `id`, `title`, `genre`, `thumbnail`, `shortDescription`, `gameUrl`,
+       `genre`, `platform`, `publisher`, `developer`, `releaseDate`, `freetogameProfileUrl`
+        FROM `pc-games` WHERE `id` = ' . $id . ';');
```
Comment (norayna) [inline src/Hydrators/GamesHydrator.php]: SQL injection vulnerability 😱

**Review by norayna — CHANGES_REQUESTED** [review]
please make changes 😇

File: src/Hydrators/GamesHydrator.php (hunk 2 above)
Comment (GenadiKozarev) [reply src/Hydrators/GamesHydrator.php]: updated

File: gameDetails.php (hunk 1 above)
Comment (GenadiKozarev) [reply gameDetails.php:9]: fixed

File: src/Hydrators/GamesHydrator.php (hunk 2 above)
Comment (norayna) [reply src/Hydrators/GamesHydrator.php]: it's still concatenating, need to use :id in the query

Comment (MrLukey) [reply src/Hydrators/GamesHydrator.php]: should be all good

**Review by norayna — APPROVED** [review]

**Pull request merged**